src/coreclr/nativeaot/Common/src/Internal/Runtime/CompilerHelpers/StartupCodeHelpers.cs

```diff
@@ -290,12 +290,7 @@ private static unsafe void RehydrateData(IntPtr dehydratedData, int length)
                         {
                             // At the time of writing this, 90% of DehydratedDataCommand.Copy cases
                             // would fall into the above specialized cases. 10% fall back to memmove.
-                            memmove(pDest, pCurrent, (nuint)payload);
-
-                            // Not a DllImport - we don't need a GC transition since this is early startup
-                            [MethodImplAttribute(MethodImplOptions.InternalCall)]
-                            [RuntimeImport("*", "memmove")]
-                            static extern unsafe void* memmove(byte* dmem, byte* smem, nuint size);
+                            Unsafe.CopyBlock(pDest, pCurrent, (uint)payload);
```

Comment:
@EgorBo would this use the managed impl of the block copying after your #98623?

Wonder whether we should just replace the whole block from line 266 onwards with that.

When I originally wrote this code in https://github.com/dotnet/runtime/pull/77884#discussion_r1023636425 I had concerns about this running extremely early at startup. If the managed impl works here, it would probably generate equivalent code with the added benefit of being even better for `--instruction-set native`.

Reply:
> @EgorBo would this use the managed impl of the block copying after your https://github.com/dotnet/runtime/pull/98623?

Correct, it will be a direct call to https://github.com/dotnet/runtime/blob/main/src/libraries/System.Private.CoreLib/src/System/SpanHelpers.ByteMemOps.cs#L253

Reply:
Oh, this is the thing called through RuntimeExports...

If we get a non-inlineable call then what this PR is doing is enough. We can't get rid of the small size special cases, the outer loop is very hot.

Reply:
If needed, we can consider exposing native memset/memmove/memset APIs to `NativeMemory.cs`